core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java

```diff
@@ -581,6 +581,19 @@ private List<TemplateParameterInfo> gatherTemplateParamInfo(List<MethodParameter
         return templateParams;
     }
 
+    private List<AnnotationInstance> effectiveParamAnnotations(MethodParameterInfo param) {
+        return param.annotations().stream().filter(ai -> {
+            String name = ai.name().toString();
+            if (name.startsWith("kotlin")) {
+                return false;
+            }
+            if (name.endsWith("NotNull")) {
```

Comment:
I wonder if you should ignore everything under `jakarta.validation.constraints.` instead.

Reply:
NotNull are super common - most tools even allow custom ones, so I'll leave the rule here

Reply:
I was wondering if we shouldn't add CheckReturnValue.

Reply:
That is only for return values,  no?

Reply:
Jakarta Validation also has `@NotBlank` which could be useful in this case. Thus my recommendation.

You can keep not null as is but I would also ignore everything under jakarta.validation.constraints.